### ISSUE TITLE
Scheduler: make KubeSchedulerConfiguration override the default plugins configuration

### ISSUE THREAD

**Issue by sanposhiho** [kubernetes/kubernetes#103956]
Hello team. 

Now, if users want to change the default plugin’s weight, they have to disable the plugin once and enable it with wanted weight on KubeSchedulerConfiguration like below. 

```yaml
      score:
        enabled:
          - name: NodePreferAvoidPods
            weight: 123  # (NodePreferAvoidPods’s default weight is 10000)
        disabled:
          - name: NodePreferAvoidPods
```

Isn't it non-simple? Users may not know which setting takes precedence, enabled or disabled. 

Is there a definite reason for this? Or, is changing the weight of the default plugin in itself a bad thing to do?

I'd like to hear your opinion about this point.

This issue may be related to #102303

/sig scheduling

## solutions

If we want to fix this, this is one of the solutions where configuration on user-specified KubeSchedulerConfiguration override the default plugins configuration.

When users add a default plugin on KubeSchedulerConfiguration (without disabled it), it gives priority to the user's configuration values. This allows the user to override the default plugin weight by setting like below.

```yaml
      score:
        enabled:
          - name: NodePreferAvoidPods
            weight: 123
```

**Comment by kerthcet** [comment]
hey @sanposhiho , can i have a knowledge of which apiversion `kubescheduler.config.k8s.io`  applied, as  `NodePreferAvoidPods` is deprecated in `kubescheduler.config.k8s.io/v1beta2`

**Comment by sanposhiho** [comment]
Sorry @kerthcet I missed.😢 This problem was already solved in #99582.

So, close this issue.
